README.md

```diff
@@ -0,0 +1,113 @@
+# Swarm Deployment
+
+* [Summary](#Summary)
+* [Description](#Description)
+* [How To](#how-to-use-module)
+* [About Us](#ABOUT-US)
+    * [Contributing](#Contributing)
+    * [License](#License)
+
+## Summary
+This repository defines Pulumi Component Resources for deploying compute instances and initialising a docker swarm.
+
+## Description
+### SSH Keys
+For automated deployments, an SSH key is generated on the deploying device, and is added to the compute instance.
+This allows the deployer to create a docker context to connect and deploy a stack to the swarm cluster.
+This can be done by adding an ssh context from local machine, i.e. `docker context create test-swarm --docker "host=ssh://deployer@{instance-ip}"`
+
+### Compute Instances
+The program creates and manages compute instances on GCP. The first instance is initialized as the Docker Swarm manager node. 
+Subsequent instances join the Swarm using a token saved in Google Secret Manager. Each instance:
+
+- Is within the subnet created by the SwarmNetwork component. The instances are assigned different zones (a, b or c)
+  based on the instance number (e.g. instance 0,3,6 etc. would be zone `a`, 1,4,7 etc. would be `b`, and 2,5,8 etc. would be `c`)
+- Has docker installed and initialises/joins docker swarm node
+- Has generated SSH key added so the deployer can easily interact with the cluster.
+
+### Network Components
+The following network components are created:
+
+- The defined swarm network.
+- A subnet with a specified CIDR range and a secondary range `172.17.0.0/16` for Docker communications.
+- Firewall rules:
+  - **Internal Swarm Communication:** Allows nodes in the Swarm to communicate with each other, accepting all tcp, udp, 
+    icmp and esp connections between instances. 
+  - **SSH Access:** Permits SSH access to the instances for specified IPs
+  - **Service Access:** If any services are exposed on the nodes, this rule will ensure they're accessible.
+
+
+## How to use module
+
+### Import module
+
+1. Create a pulumi program
+2. Include this module in requirements.txt (e.g. `swarm-deployment @ git+https://github.com/mesoform/pulumi-swarm-deployment.git`)
+3. Use Component resources in code (see example)
+
+### Cloning
+1. Clone this repository.
+2. Navigate to the directory containing the Pulumi program.
+3. Modify the configuration for the stack
```

Comment:
Might be good to add that at least these 3 parameters are required on a stack:
e.g: Pulumi.test.yaml
```
config:
  gcp:project: test-project
  swarm-deployment-gcp:docker_token_secret_name: docker-token
  swarm-deployment-gcp:name: swarm-deployment
```